### PR TITLE
Dev add sparse softmax cross entropy

### PR DESCRIPTION
**Pull request by hjchen2**

### CODE REVIEW

File: python/oneflow/test/cambricon/test_sparse_softmax_cross_entropy.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    device_type, data_type, label_type, batch_size, num_classes,
+):
+    data_type = type_name_to_flow_type[data_type]
+    label_type = type_name_to_flow_type[label_type]
+    np_labels = np.random.randint(0, num_classes, size=(batch_size,)).astype(np.int32)
+    np_logits = np.random.random((batch_size, num_classes)).astype(np.float32)
+
+    of_logits_cpu = flow.tensor(np_logits, dtype=data_type, requires_grad=True)
+    of_labels_cpu = flow.tensor(np_labels, dtype=label_type)
+    of_output_cpu = flow.nn.functional.cross_entropy(
```
Comment (BBuf) [inline python/oneflow/test/cambricon/test_sparse_softmax_cross_entropy.py:42]: 这里为啥不是直接和flow.nn.functional.sparse_softmax_cross_entropy做对比而是low.nn.functional.cross_entropy

File: oneflow/cambricon/kernels/sparse_softmax_cross_entropy_kernel.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        prob->mut_dptr()));
+  }
+
+  bool AlwaysComputeWhenAllOutputsEmpty() const override { return false; }
+};
+
+REGISTER_USER_KERNEL("sparse_softmax_cross_entropy")
+    .SetCreateFn<MluSparseSoftmaxCrossEntropyKernel>()
+    .SetIsMatchedHob((user_op::HobDeviceType() == DeviceType::kMLU)
+                     && ((user_op::HobDataType("out", 0) == DataType::kFloat)
```
Comment (BBuf) [inline oneflow/cambricon/kernels/sparse_softmax_cross_entropy_kernel.cpp:63]: 文档提到half可能造成精度损失，这个点可能需要注意一下，可能引起模型不收敛问题。
图片

File: python/oneflow/test/cambricon/test_sparse_softmax_cross_entropy.py (hunk 1 above)
Comment (BBuf) [reply python/oneflow/test/cambricon/test_sparse_softmax_cross_entropy.py:42]: 哦，看了下这里用sparse_softmax_cross_entropy也可以，因为softmax_cross_entropy可以接受类别标签tensor或者一个概率分布矩阵tensor，而sparse_softmax_cross_entropy只能接收类别标签tensor作为target

**Review by BBuf — APPROVED** [review]

File: python/oneflow/test/cambricon/test_sparse_softmax_cross_entropy.py (hunk 1 above)
Comment (hjchen2) [reply python/oneflow/test/cambricon/test_sparse_softmax_cross_entropy.py:42]: 这个测试代码原本是和pytorch cpu对比的，我给改成了和oneflow cpu，不过确实和你说的那样，这里最好还是改成sparse_softmax_cross_entropy

File: oneflow/cambricon/kernels/sparse_softmax_cross_entropy_kernel.cpp (hunk 2 above)
Comment (hjchen2) [reply oneflow/cambricon/kernels/sparse_softmax_cross_entropy_kernel.cpp:63]: 是的，这里只是把half先支持了，用不用是上层逻辑的事

File: python/oneflow/test/cambricon/test_sparse_softmax_cross_entropy.py (hunk 1 above)
Comment (hjchen2) [reply python/oneflow/test/cambricon/test_sparse_softmax_cross_entropy.py]: done
